### ISSUE TITLE
Counting game should alternate grouped and ungrouped object challenges

### ISSUE THREAD

**Issue by amanda-phet** [phetsims/number-play#126]
Currently, the challenges alternate
- ten-frame
- grouped objects
- ten-frame
- grouped objects
- etc.

I think we want something more like this
- ten-frame
- ungrouped objects
- ten-frame
- grouped objects
- ten-frame
- ungrouped objects
- ten-frame
- grouped objects
- etc.

**Comment by chrisklus** [comment]
Changes committed above.

At some point in the last month or two, @amanda-phet and I discussed this and @amanda-phet clarified that it was not imperative that ten-frame comes first every time a user re-enters the level, just that the pattern that she identified is followed. This made the implementation much simpler to do.

Ready for review by @amanda-phet. I only applied this change only to level 2 since that was the only level with grouping but it can easily be added to level 1 if desired.

**Comment by amanda-phet** [comment]
@chrisklus is looking into this as a bug.

**Comment by chrisklus** [comment]
@amanda-phet nice catch!

I fixed this in the above commits. The problem was that the the objects were not being repositioned if the challenge number didn't change from the last challenge. But, the grouping/ungrouping state was still being flipped, so that caused either ungrouped-positioned objects to have card background, or grouped-positioned cards to switch to stacks.

I also made some improvements to the spread of the random object distribution. Note how the two screenshots from https://github.com/phetsims/number-play/issues/126#issuecomment-1076596069 show the objects spread out in the top half of the play area only - this was the default positioning to make room for the (nonexistent) creator node. So they are using the full space now, and the handles don't stick out the top after being distributed like they were sometimes previously.

@amanda-phet ready for review!
